src/main/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseData.scala

```diff
@@ -161,8 +166,53 @@ class AppInfoBaseData(
       case NonFatal(e) => throw new RuntimeException(s"while retrieving last task failure for app [${app.id}]", e)
     }
   }
+
+  @SuppressWarnings(Array("all")) // async/await
+  def podStatus(podDef: PodDefinition)(implicit ec: ExecutionContext): Future[PodStatus] =
+    async { // linter:ignore UnnecessaryElseBranch
+      val now = clock.now().toOffsetDateTime
+      val instances = await(instanceTracker.specInstances(podDef.id))
```

Comment:
Rename `tasksByAppFuture` to `instancesByRunSpecFuture` and use this method here.